### ISSUE TITLE
libjlibtorrent.so has text relocations

### ISSUE THREAD

**Issue by jackwakefield** [frostwire/frostwire-jlibtorrent#85]
This produces a warning on Android devices lower than Marshmallow, but an error on Marshmallow and above.

```
java.lang.LinkageError: Look for your architecture binary instructions at: https://github.com/frostwire/frostwire-jlibtorrent
    at com.frostwire.jlibtorrent.swig.libtorrent_jni.<clinit>(libtorrent_jni.java:22)
    at com.frostwire.jlibtorrent.swig.settings_pack.<init>(settings_pack.java:71)
    at com.frostwire.jlibtorrent.SettingsPack.<init>(SettingsPack.java:31)
    at com.frostwire.jlibtorrent.Session.<init>(Session.java:84)
    at com.github.se_bastiaan.torrentstream.TorrentStream$1.run(TorrentStream.java:103)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:148)
    at android.os.HandlerThread.run(HandlerThread.java:61)
    Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: /data/app/com.x.app-1/lib/x86/libjlibtorrent.so: has text relocations
    at java.lang.Runtime.loadLibrary(Runtime.java:372)
    at java.lang.System.loadLibrary(System.java:1076)
    at com.frostwire.jlibtorrent.swig.libtorrent_jni.<clinit>(libtorrent_jni.java:17)
    at com.frostwire.jlibtorrent.swig.settings_pack.<init>(settings_pack.java:71) 
    at com.frostwire.jlibtorrent.SettingsPack.<init>(SettingsPack.java:31) 
    at com.frostwire.jlibtorrent.Session.<init>(Session.java:84) 
    at com.github.se_bastiaan.torrentstream.TorrentStream$1.run(TorrentStream.java:103) 
    at android.os.Handler.handleCallback(Handler.java:739) 
    at android.os.Handler.dispatchMessage(Handler.java:95) 
    at android.os.Looper.loop(Looper.java:148) 
    at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by gubatron** [comment]
@jackwakefield what version of jlibtorrent do you have on your gradle dependencies?

**Comment by jackwakefield** [comment]
@gubatron 1.1.0.23, I've just tested with 1.1.0.22 which doesn't throw the error.

**Comment by aldenml** [comment]
@jackwakefield I will review that, I documented the problem here https://github.com/frostwire/frostwire-jlibtorrent/issues/83#issuecomment-205423452. At this moment I don't know if it's related to the upgrade of OpenSSL (having some new assembly) or just upgrading the toolchain solve the problem. If it is possible for you, If only for a few days, you can change your targetSDK to 22 and 1.1.0.23 should work.

**Comment by aldenml** [comment]
@jackwakefield I published 1.1.0.24, this solve the problem of Android 6 in x86 arch. Additional, this is in sync with the just released libtorrent 1.1.0

**Comment by se-bastiaan** [comment]
Great work. Thanks for fixing it so fast.
